ext/mysqlnd/mysqlnd_commands.c

```diff
@@ -642,15 +642,6 @@ MYSQLND_METHOD(mysqlnd_command, handshake)(MYSQLND_CONN_DATA * const conn, const
 	conn->protocol_version	= greet_packet.protocol_version;
 	conn->server_version	= mnd_pestrdup(greet_packet.server_version, conn->persistent);
 
-	conn->greet_charset = mysqlnd_find_charset_nr(greet_packet.charset_no);
```

Comment:
MariaDB 11.3 Docker images have been patched - https://github.com/MariaDB/mariadb-docker/commit/d7a950d41e9347ac94ad2d2f28469bff74858db7

but the GH-13452 can be tested using 11.3.1-rc image - https://hub.docker.com/layers/library/mariadb/11.3.1-rc/images/sha256-f37b1ba4f15fb90679f8f85fd7300815ef4c5ff7ada668bccbdff33e2ad8ecd6

Reply:
Note can also test by placing `--character-set-collations=utf8mb4=uca1400_ai_ci` as the command on the 11.3+ image to enforce the setting.